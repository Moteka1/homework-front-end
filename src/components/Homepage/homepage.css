.homepage-container {
  max-width: 1000px;
  margin: auto;
}

.homepage-trending {
  display: flex;
  overflow-x: scroll;
}

.homepage-foundGifs {
  display: flex;
  flex-wrap: wrap;
}
