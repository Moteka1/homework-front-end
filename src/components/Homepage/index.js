import React, { Component } from 'react';
import SearchBar from '../SearchBar';
import axios from 'axios';
import GifCard from '../GifCard';
import './homepage.css';

export default class Homepage extends Component {
  state = {
    gifResults: [],
    trendingGifs: [],
    loading: true // will need to add for a loading placeholder later
  };

  // to have trending gifs need a component did mount to load API request once they are ready
  componentDidMount = async () => {
    this.trendingGiphy();
  };
  // search function takes in a term and return array of gif results
  searchGiphy = async searchTerm => {
    try {
      const limit = 25;
      const apiKey = 'KdpYtENdZIbVcvy4BGZEDKNameGAKyaw'; // process.env.REACT_APP_GIPHY_API_KEY;
      const url = `http://api.giphy.com/v1/gifs/search?q=${searchTerm}&api_key=${apiKey}&limit=${limit}`;
      const response = await axios.get(url).then(res => {
        return res.data.data; // array of objs, each obj contains gif info from the search
      });
      // update state in a way that does not mutate state
      this.setState({
        ...this.state,
        gifResults: response
      });
    } catch (error) {
      console.log(error);
    }
  };

  trendingGiphy = async () => {
    try {
      const url =
        'https://api.giphy.com/v1/gifs/trending?api_key=KdpYtENdZIbVcvy4BGZEDKNameGAKyaw';
      const response = await axios.get(url).then(res => {
        return res.data.data; // arr of objs of trending gifs
      });
      this.setState({
        ...this.state,
        trendingGifs: response
      });
    } catch (error) {
      console.log(error);
    }
  };

  render() {
    let foundGifs = this.state.gifResults.map(gif => (
      // might want gif.images.downsized.url for searched
      <GifCard
        gifSrc={gif.images.downsized.url}
        gifAlt={gif.title}
        key={gif.id}
        title={gif.title}
        username={gif.username}
        rating={gif.rating}
        giphyUrl={gif.url}
      />
    ));

    let trendingGifs = this.state.trendingGifs.map(gif => (
      <GifCard
        gifSrc={gif.images.fixed_height.url}
        gifAlt={gif.title}
        key={gif.id}
        title={gif.title}
        username={gif.username}
        rating={gif.rating}
        giphyUrl={gif.url}
      />
    ));

    return (
      <div className="homepage-container">
        {/* Navbar */}
        <h1>Find A Gif</h1>
        {/* search bar */}
        <SearchBar search={this.searchGiphy} />
        {/* trending */}
        <h3>Trending Gifs!</h3>
        <div className="homepage-trending">{trendingGifs}</div>
        {/* home page/ area to fill, maybe route to a search page*/}
        <div className="homepage-foundGifs">{foundGifs}</div>
        {/* <GifArea gifResults={this.state.gifResults} /> */}
        {/* each giphy is a card */}
      </div>
    );
  }
}
