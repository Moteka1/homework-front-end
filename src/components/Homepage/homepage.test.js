import React from 'react';
import Homepage from './index.js';
import renderer from 'react-test-renderer';

describe('<Homepage />', () => {
  it('matches the snapshot', () => {
    var tree = renderer.create(<Homepage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
