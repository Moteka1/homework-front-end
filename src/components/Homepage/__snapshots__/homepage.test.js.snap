// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> matches the snapshot 1`] = `
<div
  className="homepage-container"
>
  <h1>
    FIND A GIF
  </h1>
  <div
    className="searchbar-container"
  >
    <div
      className="searchbar-content"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          htmlFor="search"
        >
          <input
            className="searchTerm"
            id="search"
            name="searchTerm"
            onChange={[Function]}
            placeholder="Search for Gif"
            type="text"
            value=""
          />
        </label>
        <input
          className="searchbar-button"
          type="submit"
          value="Search"
        />
      </form>
    </div>
  </div>
  <h3
    className="homepage-trendin-header"
  >
    Trending Gifs!
  </h3>
  <div
    className="homepage-trending"
  />
  <h2>
    Gifs found will display below!
  </h2>
  <img
    alt="placeholder"
    className="placeholder"
    src="http://icons-for-free.com/free-icons/png/512/283043.png"
  />
  <div
    className="homepage-foundGifs"
  />
</div>
`;
